easybuild/easyblocks/a/amber.py

```diff
@@ -0,0 +1,223 @@
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for Amber, implemented as an easyblock
+
+Original author: Benjamin Roberts (The University of Auckland)
+
+Modified by Stephane Thiell (Stanford University) for Amber14
+"""
+import os
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM, MANDATORY, BUILD
+from easybuild.tools.build_log import EasyBuildError
+import easybuild.tools.environment as env
+from easybuild.tools.run import run_cmd
+from easybuild.tools.modules import get_software_root, get_software_version
+import easybuild.tools.toolchain as toolchain
+
+class EB_Amber(ConfigureMake):
+    """Easyblock for building and installing Amber"""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Extra easyconfig parameters specific to ConfigureMake."""
+        extra_vars = dict(ConfigureMake.extra_options(extra_vars))
+        extra_vars.update({
+            # 'Amber': [True, "Build Amber in addition to AmberTools", CUSTOM],
+            'patchlevels': ["latest", "(AmberTools, Amber) updates to be applied", CUSTOM],
+            # The following is necessary because some patches to the Amber update
+            # script update the update script itself, in which case it will quit
+            # and insist on being run again. We don't know how many times will
+            # be needed, but the number of times is patchlevel specific.
+            'patchruns': [1, "Number of times to run Amber's update script before building", CUSTOM],
+        })
+        return ConfigureMake.extra_options(extra_vars)
+
+    def __init__(self, *args, **kwargs):
+        super(EB_Amber, self).__init__(*args, **kwargs)
+        self.already_extracted = False
+        self.build_in_installdir = True
+
+    def extract_step(self):
+        """Only extract from the tarball if this has not already been done."""
+        if not self.already_extracted:
```

Comment:
Yes, I was referring to the `already_extracted` conditional... I guess it's a leftover of earlier attempts by @valtandor.

I'm cleaning up this easyblock a bit, so I'll chuck this and see what happens.